### PR TITLE
[refactor](stats) Persist status of analyze task to FE meta data

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

1. In the past, we use a BE table named `analysis_jobs` to persist the status of analyze jobs/tasks, however there are many flaws such as, if BE crashed analyze job/task would failed however the status of analyze job/task couldn't get updated.
2. Support `DROP ANALYZE JOB [job_id]` to delete analyze job
3. Support `SHOW ANALYZE TASK STATUS [job_id] ` to  get the task status of specific job
4. Restrict the execute condition of auto analyze, only when  the  last execution of auto analyze job finished a while ago could be executed again
5. Support analyze whole DB

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 1]
```diff
@@ -416,13 +421,16 @@ public ShowResultSet execute() throws AnalysisException {
...
+        } else if (stmt instanceof ShowAnalyzeTaskStatus) {
+            handleShowAnalyzeTaskStatus();
         } else {
             handleEmtpy();
         }
 
         return resultSet;
     }
 
+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:433]: remove this blank

File: fe/fe-core/src/main/java/org/apache/doris/persist/OperationType.java [hunk 2]
```diff
@@ -295,6 +295,15 @@ public class OperationType {
     // update binlog config
     public static final short OP_UPDATE_BINLOG_CONFIG = 425;
 
+    public static final short OP_CREATE_ANALYSIS_TASK = 435;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/persist/OperationType.java:298]: recommand use from 430 to 439

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 3]
```diff
@@ -2793,11 +2793,7 @@ analyze_stmt ::=
         }
         // Rule: If no type is specified, see if there is a specified column
         if (!properties.containsKey("analysis.type")) {
-            if ((cols == null)) {
-                properties.put("analysis.type", "INDEX");
-            } else {
-                properties.put("analysis.type", "COLUMN");
-            }
+            properties.put("analysis.type", "FUNDAMENTALS");
```
Comment (morrySnow) [inline fe/fe-core/src/main/cup/sql_parser.cup:2796]: need update docs

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisInfoBuilder.java [hunk 4]
```diff
@@ -17,24 +17,24 @@
...
+public class AnalysisInfoBuilder {
     private long jobId;
     private long taskId;
     private String catalogName;
     private String dbName;
     private String tblName;
     private Map<String, Set<String>> colToPartitions;
     private String colName;
-    private Long indexId;
+    private Long indexId = -1L;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisInfoBuilder.java:37]: why use boxing type Long?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 5]
```diff
@@ -384,53 +415,59 @@ private void createTaskForMVIdx(AnalysisTaskInfo jobInfo, Map<Long, BaseAnalysis
...
             }
         }
     }
 
-    private void createTaskForExternalTable(AnalysisTaskInfo jobInfo,
-                                            Map<Long, BaseAnalysisTask> analysisTasks,
-                                            boolean isSync) throws DdlException {
+    private void logCreateAnalysisTask(AnalysisInfo analysisInfo) {
+        Env.getCurrentEnv().getEditLog().logCreateAnalysisTasks(analysisInfo);
+        analysisTaskInfoMap.put(analysisInfo.taskId, analysisInfo);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:460]:
```suggestion
        analysisTaskInfoMap.put(analysisInfo.taskId, analysisInfo);
        Env.getCurrentEnv().getEditLog().logCreateAnalysisTasks(analysisInfo);
```

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 6]
```diff
@@ -384,53 +415,59 @@ private void createTaskForMVIdx(AnalysisTaskInfo jobInfo, Map<Long, BaseAnalysis
...
+    }
+
+    private void logCreateAnalysisJob(AnalysisInfo analysisJob) {
+        Env.getCurrentEnv().getEditLog().logCreateAnalysisJob(analysisJob);
+        analysisJobInfoMap.put(analysisJob.jobId, analysisJob);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:465]:
```suggestion
        analysisJobInfoMap.put(analysisJob.jobId, analysisJob);
        Env.getCurrentEnv().getEditLog().logCreateAnalysisJob(analysisJob);
```

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 7]
```diff
@@ -675,10 +718,57 @@ public void execute() {
...
+
+    public void dropAnalyzeJob(DropAnalyzeJobStmt analyzeJobStmt) throws DdlException {
+        AnalysisInfo jobInfo = analysisJobInfoMap.get(analyzeJobStmt.getJobId());
+        if (jobInfo == null) {
+            throw new DdlException(String.format("Analyze job [%d] not exists", jobInfo.jobId));
+        }
+        checkPriv(jobInfo);
+        long jobId = analyzeJobStmt.getJobId();
+        AnalyzeDeletionLog analyzeDeletionLog = new AnalyzeDeletionLog(jobId);
+        Env.getCurrentEnv().getEditLog().logDeleteAnalysisJob(analyzeDeletionLog);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:769]: all operation on edit log should do action first and then write edit log

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**